src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -186,7 +187,7 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
                         throw new IOException("failed to run ps");
                     }
                     List<String> pids = new ArrayList<String>();
-                    LINE: for (String line : baos.toString().split("\n")) {
+                    LINE: for (String line : baos.toString(Charset.defaultCharset().name()).split("\n")) {
                         for (Map.Entry<String,String> entry : modelEnvVars.entrySet()) {
```

Comment:
not sure is important (probably NOT) but since you re cycling on the lines you could save `Charset.defaultCharset().name()` before in a variable as you did later in the code (`final String charsetName = Charset.defaultCharset().name();`). but it's not of much computational cost anyway I think so feel free to ignore this comment